config/experimental/configsource/factory.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configsource
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+)
+
+// CreateSettings is passed to Factory.CreateConfigSource function.
+type CreateSettings struct {
```

Comment:
nit: CreateSettings sounds like a function name which starts with a verb. Would it be a good idea to rename the struct?

Reply:
I agree, but this is following the general pattern for the other components. The difference being that because they exist on the `config` package they prefix it with the component type, e.g. `config.ProcessorCreateSettings`. Right now if we follow it becomes `configsource.ConfigSourceCreateSettings`. Let's keep it as it is until it moves out of experimental.

Reply:
Probably better to keep the package "config" for the moment even for this? So that when we move this to the config we don't break too many things?

Reply:
Ok, moved the config and component related to under experimental to their own packages - had to remove "ConfigSource" from some of what would be typical names in order to avoid stutter flagged by lint.